api/CMakeLists.txt

```diff
@@ -1,161 +1,137 @@
-        SET(LIBXML_LIB ${LIBXML2_LIBRARIES} CACHE STRING "The name of the libxml2 release library.")
-    ELSE(LIBXML2_INCLUDE_DIR)
-        SET(LIBXML_INCLUDE_DIR "/usr/include" CACHE PATH "Location of libxml2 header files.")
-        # The name of the libxml2 release library.
-        SET(LIBXML_LIB "libxml2" CACHE STRING "The name of the libxml2 release library.")
-    ENDIF()
-
-ELSEIF(MINGW)
-    # The location of the libxml2 header files.
-    FIND_PACKAGE(LibXml2)
-    IF(LIBXML2_INCLUDE_DIR)
-        SET(LIBXML_INCLUDE_DIR ${LIBXML2_INCLUDE_DIR} CACHE PATH "Location of libxml2 header files.")
-        SET(LIBXML_LIB ${LIBXML2_LIBRARIES} CACHE STRING "The name of the libxml2 release library.")
-    ELSE(LIBXML2_INCLUDE_DIR)
-        SET(LIBXML_INCLUDE_DIR "${PROJECT_SOURCE_DIR}/extlibs/libxml2/installed/include/" CACHE PATH "Location of libxml2 header files.")
-        # The name of the libxml2 release library.
-        SET(LIBXML_LIB "${PROJECT_SOURCE_DIR}/extlibs/libxml2/installed/lib/libxml2" CACHE STRING "The name of the libxml2 release library.")
-    ENDIF()
-ENDIF()
-
-include_directories (${HDF5_INCLUDE_DIR} ${BEECRYPT_INCLUDE_DIR} ${LIBXML_INCLUDE_DIR}) 
 
 # The debug build will have a 'd' postfix
-SET(CMAKE_DEBUG_POSTFIX  "d")
-
-if(WIN32)
-	#Preset install dir
-	IF(CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT)
-		SET(CMAKE_INSTALL_PREFIX "${PROJECT_SOURCE_DIR}/installed" CACHE PATH "CMAKE_INSTALL_PREFIX" FORCE)
-	ENDIF(CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT)
-endif()
-
-#Use the version 1 interface for HDF5
-add_definitions(-D H5_USE_16_API)
-
-IF(MSVC)
-
-    #ignore some warnings generated by windows.
-    add_definitions(/D_CRT_SECURE_NO_WARNINGS /D _CRT_NONSTDC_NO_DEPRECATE)
-    
-	#if we are building a shared library, then we need to define these symbols.
-	IF(BUILD_SHARED_LIBS)
-		add_definitions(/D BAG_DLL /D BAG_EXPORTS)
-	ENDIF(BUILD_SHARED_LIBS)
-	
-	#if we are building against a shared HDF5, then we need to define this symbol.
-	IF(BUILD_WITH_SHARED_HDF5)
-		add_definitions(/D _HDF5USEDLL_)
-	ENDIF(BUILD_WITH_SHARED_HDF5)
-	
-ENDIF(MSVC)
-
-#List all of the bag api source files.
-set (BAG_SOURCE_FILES 
- bag_attr.c
- bag_crypto.c
- bag_hdf.c
- bag_metadata.cpp
- bag_metadata_export.cpp
- bag_metadata_import.cpp
- bag_legacy.c
- bag_opt_group.c
- bag_opt_surfaces.c
- bag_reference_system.cpp
- bag_surface_correct.c
- bag_surfaces.c
- bag_tracking_list.c
- crc32.c
- onscrypto.c)
+set(CMAKE_DEBUG_POSTFIX "d")
+
+set(BAG_SOURCE_FILES
+    bag.cpp
+    bag_compoundlayer.cpp
+    bag_dataset.cpp
+    bag_descriptor.cpp
+    #bag_hdf.c
+    bag_interleavedlayer.cpp
+    bag_interleavedlayerdescriptor.cpp
+    bag_layer.cpp
+    bag_layerdescriptor.cpp
+    bag_legacy_crs.cpp
+    bag_metadata.cpp
+    bag_metadata_export.cpp
+    bag_metadata_import.cpp
+    bag_metadatatypes.cpp
+    #bag_opt_group.c                TODO
```

Comment:
What is the intent of these commented .c files?

Reply:
At the time, I was not sure if these would be reused, or how.  Now I keep them around to refer to them and make sure I create the cpp version correctly.  The .c files will be removed in the near future, and any references to them.